js/main.js

```diff
@@ -100,30 +100,50 @@ document.addEventListener('DOMContentLoaded', function () {
   }
 });
 
+var $modalContainer = document.querySelector('.modal-container');
+
 document.addEventListener('click', function (event) {
   if (!event.target.classList.contains('view-changer')) {
     return;
   }
-  changeView(event.target.getAttribute('data-view'));
   if (event.target.innerText === 'NEW') {
     $newImgURL.setAttribute('src', 'images/placeholder-image-square.jpg');
     $formEnterNew.reset();
     $h1NewAndEditEntry.textContent = 'New Entry';
+    $deleteButton.classList.add('hidden');
+    changeView(event.target.getAttribute('data-view'));
+  } else if (event.target.innerText === 'Delete Entry') {
+    $modalContainer.classList.remove('hidden');
+  } else if (event.target.innerText === 'CANCEL') {
+    $modalContainer.classList.add('hidden');
+  } else if (event.target.innerText === 'CONFIRM') {
```

Comment:
Determining behavior by the text content is not a good idea, because if a graphic designer ever swoops in and decides that "Delete" fits better with their new design than "Delete Entry", suddenly the app doesn't work anymore. You should instead use a data attribute or something to that effect